ui/raidboss/data/06-ew/raid/p12s.ts

```diff
@@ -465,6 +490,328 @@ const triggerSet: TriggerSet<Data> = {
+        lightTilt: {
+          en: 'Soak dark tower',
+        },
+        darkTilt: {
+          en: 'Soak light tower',
+        },
+      },
+    },
+    // In Engravement 2 (Superchain 1), all supports or  DPS will receive lightTilt and darkTilt (2 each).
+    // All 4 also receive Heavensflame Soul.
+    // The other role group will receive lightTower, darkTower, lightBeam, and darkBeam.
+    // To resolve the Beams during the 2nd orb, lightBeam needs to stack with darkTower and both darkTilts, and vice versa.
+    // After the 3rd orb, lightTower and darkTower will drop their towers, and  darkBeam and lightBeam (respectively) will soak them.
+    // The four Heavensflame players all simultaneously need to spread to drop their AoEs.
+    // Debuffs do change based on mechanic resolution, which can complicate things:
+    // - When a lightTilt player soaks a dark beam, their debuff will change to darkTilt, and vice versa.
+    // - Once the beams detonate, the lightBeam debuff disappears and is replaced with lightTilt (same with dark).
+    // So only use the initial debuff to resolve the mechanic, and use a long suppress to avoid incorrect later alerts.
+    {
+      id: 'P12S Engravement 2 Beam Soak',
+      type: 'GainsEffect',
+      netRegex: { effectId: Object.values(engravementIdMap) },
+      condition: (data, matches) => data.engravementCounter === 2 && data.me === matches.target,
+      delaySeconds: 5, // wait until first orb resolves so as not to confuse
+      durationSeconds: 7, // keep active until beams go off
+      suppressSeconds: 30,
+      response: (_data, matches, output) => {
+        // cactbot-builtin-response
+        output.responseOutputStrings = {
+          lightBeam: {
+            en: 'Light Beam (stack with Dark group)',
+          },
+          darkBeam: {
+            en: 'Dark Beam (stack with Light group)',
+          },
+          lightTower: {
+            en: 'Light Tower (stack with light group)',
+          },
+          darkTower: {
+            en: 'Dark Tower (stack with dark group)',
+          },
+          lightTilt: {
+            en: 'Stack (light group) - spread later',
+          },
+          darkTilt: {
+            en: 'Stack (dark group) - spread later',
+          },
+        };
+
+        const engraveLabel = engravementLabelMap[matches.effectId];
+        if (engraveLabel === undefined)
+          return;
+        const engraveOutputStr = output[engraveLabel]!();
+        if (engraveOutputStr === undefined)
+          return;
+        if (engravementBeamIds.indexOf(matches.effectId) !== -1)
+          return { alarmText: engraveOutputStr };
+        return { alertText: engraveOutputStr };
+      },
+    },
+    // darkTower/lightTower are 20s, but lightBeam/darkBeam are shorter and swap to lightTilt/darkTilt before the mechanic resolves.
+    // So use a fixed delay rather than one based on effect duration.
+    {
+      id: 'P12S Engravement 2 Tower Drop/Soak Reminder',
```

Comment:
Should this be combined with the In => Out call before? e.g. `In => Out => Spread` or `In => Out => Dark Tower` then `Out => Drop Dark Tower` reminder?

Reply:
There's more wonky timing here.  If it's out->in, the tower drop is almost immediately after the donut telegraph disappears, but if it's in->out, it's like 3-4 seconds before the tower drop after the out resolves.

From testing in-game, basing the tower/soak/spread alert on the effect duration felt like the right amount of warning and wasn't confusing (at least to me) while the in/out was happening.  But if you do want to add a reminder for the second half of the mechanic, I think I'd want to test a little more in-game to see if I can find a good timing that works for both in->out and out->in.

Reply:
I guess I'm suggesting to add the reminder into the in/out call (so it says `In => Out => Spread` or whatever); this call happens way earlier, so there shouldn't be any timing issues?

I think it's a separate issue for different timings here, re: in/out.  There's no "out" call after the "in" goes off right now.  I think if that existed and we wanted to combine them then that'd be one with timing issues.

Am I misunderstanding?

Reply:
> I guess I'm suggesting to add the reminder into the in/out call (so it says `In => Out => Spread` or whatever); this call happens way earlier, so there shouldn't be any timing issues?
> 
> I think it's a separate issue for different timings here, re: in/out. There's no "out" call after the "in" goes off right now. I think if that existed and we wanted to combine them then that'd be one with timing issues.
> 
> Am I misunderstanding?

I think it was the second half of your suggestion about doing the `Out => Drop Dark Tower` that tripped me up. 

No issue with adding the => Spread/Soak/Drop to the end of the in/out call as you suggest.  But that trigger fires 7 seconds before the (current) Soak/Drop/Spread triggers fire, and 11 seconds before those soak/drop/spread mechanics resolve.  So I think we'd still need a subsequent alert closer to the actual debuff going off.  Right now, that subsequent one (in this PR) just calls Soak/Drop/Spread, so are you suggesting that it should include the second-half of the in->out direction as well? (In which case I think we'd need to move it earlier and extend the duration)?

Reply:
Ah yeah, I'd say leave that second half as a TODO for later, especially if there's timing issues and also because there's not an `Out` call right now to extend into `Out => Drop Dark Tower` so we'd have to add that as well.

So, mostly just suggesting the first half, to add it to the in/out call.